app/models/rule_identifier.rb

```diff
@@ -7,11 +7,20 @@ class RuleIdentifier < ApplicationRecord
   validates :label, presence: true
   validates :system, presence: true
 
-  def self.from_openscap_parser(op_rule_identifier, rule_id)
-    if op_rule_identifier.label # rubocop:disable Style/GuardClause
-      find_or_initialize_by(label: op_rule_identifier.label,
-                            system: op_rule_identifier.system,
-                            rule_id: rule_id)
+  class << self
+    def from_openscap_parser(op_rule_identifier, rule_id)
+      if op_rule_identifier.label # rubocop:disable Style/GuardClause
+        find_or_initialize_by(label: op_rule_identifier.label,
+                              system: op_rule_identifier.system,
+                              rule_id: rule_id)
+      end
+    end
+
+    def preexisting_from_oscap_parser(new_rules)
+      ::RuleIdentifier.where(
+        label: new_rules.map { |r| r.op_source.identifier.label },
+        rule_id: new_rules.map(&:id)
+      )
```

Comment:
This query is not correct and should be updated to use `RuleIdentifier.where(label: 'one label', rule_id: 'one rule id').or(RuleIdentifier.where(...)).or(...)` (constructed with `inject` like I showed before)